homeassistant/components/modbus/base_platform.py

```diff
@@ -105,21 +107,23 @@ def async_remote_start(self) -> None:
         self.async_write_ha_state()
 
     @callback
-    def async_remote_stop(self) -> None:
+    def async_hold(self, update=True) -> None:
         """Remote stop entity."""
+        if self._cancel_call:
+            self._cancel_call()
+            self._cancel_call = None
         if self._cancel_timer:
             self._cancel_timer()
             self._cancel_timer = None
-        self._attr_available = False
-        self.async_write_ha_state()
+        if update:
+            self._attr_available = False
+            self.async_write_ha_state()
 
     async def async_base_added_to_hass(self):
         """Handle entity which will be added."""
-        self.async_remote_start()
-        async_dispatcher_connect(self.hass, SIGNAL_STOP_ENTITY, self.async_remote_stop)
-        async_dispatcher_connect(
-            self.hass, SIGNAL_START_ENTITY, self.async_remote_start
-        )
+        self.async_run()
+        async_dispatcher_connect(self.hass, SIGNAL_STOP_ENTITY, self.async_hold)
+        async_dispatcher_connect(self.hass, SIGNAL_START_ENTITY, self.async_run)
```

Comment:
I know this is existing code, but we should really unsub these methods when the entity is removed.
```suggestion
        self.async_on_remove(
            async_dispatcher_connect(self.hass, SIGNAL_STOP_ENTITY, self.async_hold)
        )
        self.async_on_remove(
            async_dispatcher_connect(self.hass, SIGNAL_START_ENTITY, self.async_run)
        )
```

Reply:
makes sense.